### PR TITLE
Kong API, schema, and validation updates

### PR DESCRIPTION
**Pull request by shaynevanasperen**

### CODE REVIEW

File: src/Kongverge/DTOs/KongConsumer.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace Kongverge.DTOs
+{
+    public sealed class KongConsumer : KongObject, IKongEquatable<KongConsumer>, IValidatableObject
+    {
+        public static readonly string ObjectName = "consumer";
+
+        protected override string[] ToStringSegments => throw new System.NotImplementedException();
+
+        public override object GetMatchValue()
```
Comment (TomFreeman) [inline src/Kongverge/DTOs/KongConsumer.cs:14]: I'm not a huge fan of this pattern of providing a placeholder for where the implementation will go, without an implementation.

It violates my preference for YAGNI, with the only value being future implementers will know where the implementation was intended to go.

However, it's not a hill I'm going to die on.

**Review by Gadam8 — APPROVED** [review]
LGTM, but there are so many files to review I can't remember my own name anymore...

File: src/Kongverge/Services/KongAdminWriter.cs [hunk 2]
```diff
@@ -45,13 +45,19 @@ public async Task DeleteRoute(string routeId)
...
-            var updated = JsonConvert.DeserializeObject<KongPlugin>(responseBody);
-            plugin.Id = updated.Id;
+            var added = JsonConvert.DeserializeObject<KongPlugin>(responseBody);
+            plugin.Id = added.Id;
+        }
+
+        public async Task UpdatePlugin(KongPlugin plugin)
+        {
+            var content = plugin.ToJsonStringContent();
+            await HttpClient.PatchAsync($"/plugins/{plugin.Id}", content);
```
Comment (TomFreeman) [inline src/Kongverge/Services/KongAdminWriter.cs:60]: y u no do Put anymore Kong?

**Review by TomFreeman — APPROVED** [review]

File: src/Kongverge/DTOs/KongConsumer.cs (hunk 1 above)
Comment (shaynevanasperen) [reply src/Kongverge/DTOs/KongConsumer.cs:14]: It was only to be "complete" with the schema validation changes, which have to directly reference these types, here: https://github.com/justeat/kongverge/pull/66/files#diff-db0a6ce6f1359507d7650c46d488aacaR309-R312

Comment (shaynevanasperen) [reply src/Kongverge/DTOs/KongConsumer.cs:14]: But otherwise I would normally agree with you on that.

**Review by charlie-haley — APPROVED** [review]

**Pull request merged**